tests/subsurfaces.cpp

```diff
@@ -508,6 +508,70 @@ TEST_P(SubsurfaceTest, subsurface_of_a_subsurface_handled)
                     wl_fixed_from_int(pointer_y_2 - subsurface_top_y - subsurface_y))));
 }
 
+TEST_P(SubsurfaceTest, subsurface_moves_under_input_device_once)
+{
+    int const input_x = surface_x + 10, input_y = surface_y + 5;
+    int const subsurface_x = -23, subsurface_y = -17;
+
+    input_device->to_screen_position(input_x, input_y);
+    client.roundtrip();
+
+    ASSERT_THAT(input_device->focused_window(), Eq((wl_surface*)subsurface)) << "precondition failed";
```

Comment:
Non-blocking nit: I don't think “precondition failed” is a useful label for `ASSERT_THAT`. It already appears in the output, and we only use `ASSERT_THAT` for precondition failures.

Reply:
> We only use `ASSERT_THAT` for precondition failures

While this is generally my policy, I'm not sure it is used consistently throughout Mir. It's also easy to miss that the failed thing is an assert and not expect. Especially if someone isn't used to gtest or our way of using gtest, immediate seeing "precondition failed" right in the output might save them some time.